powerdns_recursor/datadog_checks/powerdns_recursor/powerdns_recursor.py

```diff
@@ -126,6 +126,37 @@ def check(self, instance):
                     self.gauge('powerdns.recursor.{}'.format(stat['name']), float(stat['value']), tags=tags)
                 elif stat['name'] in PowerDNSRecursorCheck.RATE_METRICS_V4:
                     self.rate('powerdns.recursor.{}'.format(stat['name']), float(stat['value']), tags=tags)
+        self._collect_metadata(config)
+
+    def _collect_metadata(self, config):
+        fallback_url = "http://{}:{}/api".format(config.host, config.port)
+        if config.version == 4:
+            url = fallback_url
+        else:
+            url = "http://{}:{}/servers/localhost/statistics".format(config.host, config.port)
+
+        try:
+            request = self.http.get(url)
+            request.raise_for_status()
+        except Exception as e:
+            try:
+                if fallback_url is url:
+                    raise
+                request = self.http.get(fallback_url)
+                request.raise_for_status()
```

Comment:
I noticed this logic is duplicated, could we introduce a helper function to handle this fallback logic check?